update/v1alpha1/types_update_status_health.go

```diff
@@ -0,0 +1,129 @@
+	// +kubebuilder:validation:MaxLength=4096
+	Description string `json:"description,omitempty"`
+}
+
+// InsightImpactLevel describes the severity of the impact the reported condition has on the cluster or update
+// +kubebuilder:validation:Enum=Unknown;Info;Warning;Error;Critical
+type InsightImpactLevel string
+
+const (
+	// UnknownImpactLevel is used when the impact level is not known
+	UnknownImpactLevel InsightImpactLevel = "Unknown"
+	// info should be used for insights that are strictly informational or even positive (things go well or
+	// something recently healed)
+	InfoImpactLevel InsightImpactLevel = "Info"
+	// warning should be used for insights that explain a minor or transient problem. Anything that requires
+	// admin attention or manual action should not be a warning but at least an error.
+	WarningImpactLevel InsightImpactLevel = "Warning"
+	// error should be used for insights that inform about a problem that requires admin attention. Insights of
+	// level error and higher should be as actionable as possible, and should be accompanied by links to documentation,
+	// KB articles or other resources that help the admin to resolve the problem.
+	ErrorImpactLevel InsightImpactLevel = "Error"
+	// critical should be used rarely, for insights that inform about a severe problem, threatening with data
+	// loss, destroyed cluster or other catastrophic consequences. Insights of this level should be accompanied by
+	// links to documentation, KB articles or other resources that help the admin to resolve the problem, or at least
+	// prevent the severe consequences from happening.
+	CriticalInfoLevel InsightImpactLevel = "Critical"
+)
+
+// InsightImpactType describes the type of the impact the reported condition has on the cluster or update
+// +kubebuilder:validation:Enum=None;Unknown;API Availability;Cluster Capacity;Application Availability;Application Outage;Data Loss;Update Speed;Update Stalled
+type InsightImpactType string
+
+const (
+	NoneImpactType                    InsightImpactType = "None"
+	UnknownImpactType                 InsightImpactType = "Unknown"
+	ApiAvailabilityImpactType         InsightImpactType = "API Availability"
+	ClusterCapacityImpactType         InsightImpactType = "Cluster Capacity"
+	ApplicationAvailabilityImpactType InsightImpactType = "Application Availability"
+	ApplicationOutageImpactType       InsightImpactType = "Application Outage"
+	DataLossImpactType                InsightImpactType = "Data Loss"
+	UpdateSpeedImpactType             InsightImpactType = "Update Speed"
+	UpdateStalledImpactType           InsightImpactType = "Update Stalled"
+)
+
+// InsightRemediation contains information about how to resolve or prevent the reported condition
+type InsightRemediation struct {
+	// reference is a URL where administrators can find information to resolve or prevent the reported condition
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=uri
```

Comment:
Use CEL to validate that strings are valid URLs

Reply:
Will try to find an example of that